uqcsbot/scripts/urban.py

```diff
@@ -0,0 +1,43 @@
+from uqcsbot import bot, Command
+from uqcsbot.util.status_reacts import loading_status
+import requests
+
+
+URBAN_API_ENDPOINT = 'http://api.urbandictionary.com/v0/define'
+URBAN_USER_ENDPOINT = 'https://www.urbandictionary.com/define.php'
+
+
+@bot.on_command('urban')
+@loading_status
+def handle_urban(command: Command) -> None:
+    """
+    `!urban <PHRASE>` - Looks a phrase up on Urban Dictionary.
+    """
+    # Check for search phase
+    if not command.has_arg():
+        bot.post_message(command.channel, '> Usage: `!urban <SEARCH_PHRASE>`')
+        return
+    search_term = command.arg
+
+    # Attempt to get definitions from the Urban Dictionary API.
+    http_response = requests.get(URBAN_API_ENDPOINT, params={'term': search_term})
+    if http_response.status_code != 200:
+        bot.post_message(command.channel, 'There was an error accessing the Urban Dictionary API.')
+        return
+    results = http_response.json()
+
+    # If search phrase is not found, notify user.
+    if results['result_type'] != 'exact':
+        bot.post_message(command.channel, f'> No results found for {search_term}. ¯\\_(ツ)_/¯')
+        return
+
+    # Get best definition (by most 'thumbs up') and construct response.
+    sorted_definitions = sorted(results['list'], key=lambda definition: definition['thumbs_up'], reverse=True)
+    best_definition = sorted_definitions[0]
+    response_example = best_definition.get('example', '').replace('\r\n', '\n> ')  # Format example with block quotes.
```

Comment:
I finally understand `.replace('\r\n', '\n> ')`, it puts each example in its own block quote. Maybe just have something like:
```
examples = best_definition.get('example', '').split('\r\n')
response_examples = '\n'.join(f'> {example}' for example in examples)
```
might not be as sleek, but infinitely more readable

Reply:
It's just the one example, but I had a choice between formatting it with '>>>' and sending the more examples part as a second message (in order for it to not be in the block quote), or prepending '>' on each line of the example.

Reply:
That would still work though. 😁